File: android/app/src/main/java/fr/xebia/magritte/LanguageActivity.java
package fr.xebia.magritte;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class LanguageActivity extends AppCompatActivity {

    public static final String LANGUAGE_CHOICE = "LANGUAGE_CHOICE";

    public static final int LANG_FR = 1;
    public static final int LANG_CN = 2;
    public static final int LANG_IT = 3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_language);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.choice_fr)
    public void onChoiceFranceClicked() {
        Intent intent = new Intent(this, LevelActivity.class);
        intent.putExtra(LANGUAGE_CHOICE, LANG_FR);
        startActivity(intent);
    }

    @OnClick(R.id.choice_cn)
    public void onChoiceChinaClicked() {
        Intent intent = new Intent(this, LevelActivity.class);
        intent.putExtra(LANGUAGE_CHOICE, LANG_CN);
        startActivity(intent);
    }

    @OnClick(R.id.choice_it)
    public void onChoiceItalyClicked() {
        Intent intent = new Intent(this, LevelActivity.class);
        intent.putExtra(LANGUAGE_CHOICE, LANG_IT);
        startActivity(intent);
    }
}

File: android/app/src/main/java/fr/xebia/magritte/language/LanguageActivity.kt
package fr.xebia.magritte.language

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import fr.xebia.magritte.*
import fr.xebia.magritte.level.LevelActivity
import kotlinx.android.synthetic.main.activity_language.*

class LanguageActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_language)

        choice_fr.setOnClickListener { onLanguageClicked(LANG_FR) }
        choice_cn.setOnClickListener { onLanguageClicked(LANG_CN) }
        choice_it.setOnClickListener { onLanguageClicked(LANG_IT) }
    }

    private fun onLanguageClicked(code: Int) {
        val intent = Intent(this, LevelActivity::class.java)
        intent.putExtra(LANGUAGE_CHOICE, code)
        startActivity(intent)
    }
}
